### PR TITLE
fix(gomod): update package version

### LINKED ISSUE #4899: Upgrading a package with no /v* when one is availabel should add /v to the updated line
**What Renovate type are you using?**

Running with npm 

**Describe the bug**

When running renovate against a project that used 
`github.com/caarlos0/env`

provided that a higher version is available
`github.com/caarlos0/env/6`

Perform the update. 

**Did you see anything helpful in debug logs?**
As far as I understand this would if an explicit /v0|1 is set

**To Reproduce**

npm run against a project with that dependency

I'm not sure if this is the solution or if there is a reason why it shouldn't be done? I've raised a PR with a possible change that fixes this.

### PR DESCRIPTION
**Pull request by mangas**
Hopefully fixes #4899

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
@mangas Please sign the CLA (click on image) and fix the test.
image

**Review by viceice — COMMENTED** [review]
@rarkins LGTM, but we need someone who uses go for some feedback.

File: lib/manager/gomod/update.ts [hunk 1]
```diff
@@ -77,6 +77,8 @@ export function updateDependency(
         upgrade.newMajor > 1 &&
         !newLine.includes(`/v${upgrade.newMajor}`)
       ) {
+        // If package has no version, pin to latest one.
+        newLine = newLine.replace(depName, depName + '/v' + upgrade.newMajor);
```
Comment (ViceIce) [inline lib/manager/gomod/update.ts]: @mangas Is there any documentation for the new suffix handling for go modules? Does this work for all go modules?

File: lib/manager/gomod/update.ts [hunk 2]
```diff
@@ -94,7 +96,12 @@ export function updateDependency(
       }
     }
     if (lineToChange.endsWith('+incompatible')) {
-      newLine += '+incompatible';
+      let toAdd = '+incompatible';
+
+      if (upgrade.updateType === 'major' && upgrade.newMajor >= 2) {
+        toAdd = '';
+      }
+      newLine += toAdd;
```
Comment (ViceIce) [inline lib/manager/gomod/update.ts]: @mangas Any documentation for that?

File: lib/manager/gomod/update.ts (hunk 1 above)
Comment (mangas) [reply lib/manager/gomod/update.ts]: Renovate already handles the /v suffix, in this case just doesn't handle updates from no suffix to the latest suffix which means the run fails and PR doesn't get generated

Comment (mangas) [reply lib/manager/gomod/update.ts]: https://github.com/golang/go/wiki/Modules#what-are-some-implications-of-tagging-my-project-with-major-version-v0-v1-or-making-breaking-changes-with-v2

Comment (mangas) [reply lib/manager/gomod/update.ts]: This link has all the info about releases and semvers

File: lib/manager/gomod/update.ts (hunk 2 above)
Comment (mangas) [reply lib/manager/gomod/update.ts]: Same link has above. "incompatible" is used for when a package has not been migrated to go mod and or there is a version higher than 2

Comment (mangas) [reply lib/manager/gomod/update.ts]: In this case if we;'re doing a major update, if i understand this correctly, the tag is no longer needed

Comment (mangas) [reply lib/manager/gomod/update.ts]: https://github.com/golang/go/wiki/Modules#can-a-module-consume-a-v2-package-that-has-not-opted-into-modules-what-does-incompatible-mean

**Review by viceice — APPROVED** [review]

**Pull request merged**